dashboard/app/controllers/scripts_controller.rb

```diff
@@ -306,8 +307,16 @@ def get_rollup_resources
     return nil
   end
 
-  private def set_unit_by_name
-    @script = get_unit_by_name
+  private def set_unit
```

Comment:
Previously, the Unit was looked up using just the ID in the `/s/:id` path. Now we also need to look up the Unit in `/courses/:course_name/units/:unit_position` paths.